Parsers/ASimAuthentication/Parsers/vimAuthenticationCiscoIse.yaml

```diff
@@ -0,0 +1,125 @@
+  let freason_incorrect_key = dynamic([]);        //reserved for future updates
+  let freason_acc_expired = dynamic(["24414","24489"]);
+  let freason_passwd_expired = dynamic(["24054","24055","24056","24407","24484"]);
+  let freason_user_locked = dynamic(["60082","24415","24490"]);
+  let freason_user_disabled = dynamic(["51005","51006","51007","86011","86014","24409","24486"]);
+  let freason_policy_violation = dynamic(["51004","51008","22091","24057","24410","24418","24487","24496","24508","24704","24712"]);
+  let freason_session_expired = dynamic(["51003","24454","24547"]);
+  let freason_other = dynamic(["51009","51022","51106","60076","60077","61013","61015","51025","10007","22000","5400","5401","5412","5418","5448","86020","24021","24338","24518","24613","24705","24706","24707","24814","24890","24717","89157","89159","33103","11003","11039","11052","11813","11815","11823","12006","12208","12307","12308","12507","12529","12613","12615","12623","12706","12707","12976","60135","60136"]);
+  let CiscoISEAuthParser=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), targetusername_has:string="*", disabled:bool=false){
+    Syslog
+    | where not(disabled)
+  // ************************** <Prefilterring> ******************************
+    | where 
+    (isnull(starttime) or TimeGenerated >= starttime)
+    and (isnull(endtime) or TimeGenerated <= endtime)
+    and (targetusername_has=='*' or SyslogMessage has targetusername_has)
+  // ************************** </Prefilterring> *****************************
+    | where ProcessName has_any ("CISE", "CSCO")
+    | extend EventOriginalType = extract(@".*\d{10}\s(\d{4,5})\s.*", 1, SyslogMessage)
+    | where EventOriginalType in ("25104","25105","25106","25112","51000","51001","51002","51003","51004","51005","51006","51007","51008","51009","51020","51021","51022","51106","60075","60076","60077","60078","60080","60081","60082","60135","60136","60204","60205","60206","61012","61013","61014","61015","60507","51025","61076","61077","10003","10004","10005","10006","10007","22000","22004","22028","22037","22040","22091","5400","5401","5412","5418","5447","5448","86010","86011","86014","86020","24015","24020","24021","24022","24050","24054","24055","24056","24057","24337","24338","24402","24403","24406","24407","24408","24409","24410","24414","24415","24418","24454","24470","24484","24485","24486","24487","24489","24490","24491","24492","24496","24505","24508","24518","24547","24612","24613","24614","24639","24704","24705","24706","24707","24708","24709","24712","24814","24853","24854","24860","24890","24716","24717","24719","89157","89159","89160","33102","33103","5200","5201","5231","11002","11003","11039","11052","11812","11813","11814","11815","11823","11824","12005","12006","12208","12306","12307","12308","12506","12507","12528","12529","12612","12613","12614","12615","12623","12624","12705","12706","12707","12854","12975","12976","11700")
+    | extend 
```

Comment:
All string fields should come after the parsing of fields. Follow the rule of "Filter -> Parse -> Map".

Reply:
here we parse before filter to be able to retrieve correct `EventOriginalType`. If we use `SyslogMessage has_any (<all_ids>)` statement we can get unnecessary events which has these values in wrong place and not indicating `event_id `.  If this still not applicable then, please, recommend how can we handle this issue according to best practices

Reply:
I am referring to the fields like EventVendor, EventProduct. These should come after all the filters and parsing. This would ensure better performance as data size would be lowered.

Reply:
fields moved to the end of the parser